build/build.go

```diff
@@ -683,8 +683,9 @@ func Build(ctx context.Context, drivers []DriverInfo, opt map[string]Options, do
 				resp[k] = res[0]
 				respMu.Unlock()
 				if len(res) == 1 {
+					digest := res[0].ExporterResponse["containerimage.digest"]
 					if opt.ImageIDFile != "" {
```

Comment:
Maybe error out if digest is empty

Reply:
The previous code would have written an empty file to `--iidfile` in this case, changing that to an error would be an externally visible change.

In fact, looking at the code now, none of the changes in this file are actually needed, they're just a refactor due to me originally printing the digest here (and hence pulling the digest out for printing/writing to the `--iidfile`).

Reply:
Maybe best in this case would be to always set `-o type=image` if image id is to be exported, either with `iidfile`, or `q`. And error if another conflicting output is already set. This can be follow-up though as strictly related.